Multisig/Logic/Models/TokenRegistry/TokenRegistry.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  TokenRegistry.swift
+//  Multisig
+//
+//  Created by Dmitry Bespalov on 09.07.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+
+class TokenRegistry {
+
+    private var hardcoded = HardcodedTokenStore()
+    private var cache = InMemoryTokenStore()
+    private var backend = BackendTokenStore()
+    private var blockchain = BlockchainTokenStore()
+
+    // added for thread safety
+    private var queue = DispatchQueue(label: "SerialTokenRegistry")
+
+    func load(completion: @escaping () -> Void = {}) {
+        dispatchPrecondition(condition: .notOnQueue(queue))
+        queue.async { [unowned self] in
+            let tokens = self.backend.tokens()
+            tokens.forEach { self.cache.add($0) }
+            completion()
+        }
+    }
+
+    func token(address: Address) -> Token? {
+        dispatchPrecondition(condition: .notOnQueue(queue))
+
+        var result: Token?
+        queue.sync { [unowned self] in
+            if let token = self.hardcoded.token(address: address) {
+                result = token
```

Comment:
here we will always call network for Ether

Reply:
no, this is false, it will not call network